syntax = "proto3";

package inventory;

import "type/uuid.proto";
import "google/protobuf/timestamp.proto";

message ItemRequest {
	uuid id = 1;
	string code = 2;
	string name = 3;
	string description = 4;
	uint64 on_sale = 5;
	uint64 position = 6;
	double price = 7;
	uint64 track_serial = 8;
	string unit = 9;
	string status = 10;
	google.protobuf.Timestamp created_at = 11;
	google.protobuf.Timestamp updated_at = 12;
}

message ItemResponse {
	uuid id = 1;
	string code = 2;
	string name = 3;
	string description = 4;
	uint64 on_sale = 5;
	uint64 position = 6;
	double price = 7;
	uint64 track_serial = 8;
	string unit = 9;
	string status = 10;
	google.protobuf.Timestamp created_at = 11;
	google.protobuf.Timestamp updated_at = 12;
}

service Item {
	rpc GetItem(ItemRequest) returns (ItemResponse) {};
	rpc GetItems(ItemRequest) returns (stream ItemResponse) {};
}
